tensorrec/eval.py

```diff
@@ -59,6 +59,77 @@ def recall_at_k(model, test_interactions, k=10, user_features=None, item_feature
     return hit / retrieved
 
 
+def _idcg(hits, k=10, ctype="binary"):
+
+    if ctype == "binary":
+        arg = min(hits, k+1)
+        idgc = np.sum(1/np.log2(np.arange(arg) + 2))  # arange index from 0
+    elif ctype == "scalar":
+        sorted = hits[np.argsort(-hits)][:min(len(hits), k)]
+        idgc = np.sum(sorted/np.log2(np.arange(len(sorted)) + 2))
+    else:
+        raise ValueError("Invalid IDCG calculation type")
+
+    return idgc
+
+
+def normalized_discounted_cumulative_gain(model, test_interactions, k=10,
```

Comment:
For the sake of brevity, and for clarity since its an @k metric, we may want to name this `ndcg_at_k`. What do you think?

Reply:
:+1: